gateway/managers/syncman/store_kube.go

```diff
@@ -43,123 +43,55 @@ func NewKubeStore(clusterID string) (*KubeStore, error) {
+	return &KubeStore{clusterID: clusterID, kube: kube, projectsConfig: new(config.Config)}, nil
 }
 
 // Register registers space cloud to the kube store
 func (s *KubeStore) Register() {
 	// kubernetes will handle this automatically
 }
 
-func onAddOrUpdateAdminConfig(obj interface{}, clusters []*config.Admin) {
+func onAddOrUpdateResource(eventType string, obj interface{}) (string, string, interface{}) {
 	configMap := obj.(*v1.ConfigMap)
-	clusterJSONString, ok := configMap.Data["cluster"]
+	resourceID, ok := configMap.Data["id"]
 	if !ok {
-		_ = helpers.Logger.LogError(helpers.GetRequestID(context.TODO()), "Unable to watch kube store admin config as field (cluster) not found in config map", nil, nil)
-		return
+		_ = helpers.Logger.LogError(helpers.GetRequestID(context.TODO()), fmt.Sprintf("%s event occured on resource config map, but (id) field was not found in config map data", eventType), nil, nil)
+		return "", "", nil
 	}
-
-	if err := json.Unmarshal([]byte(clusterJSONString), clusters[0]); err != nil {
-		_ = helpers.Logger.LogError(helpers.GetRequestID(context.TODO()), "Unable to unmarshal config map data while watching kube store admin config", nil, nil)
-		return
-	}
-	if clusters[0].ClusterConfig == nil {
-		clusters[0].ClusterConfig = getDefaultAdminConfig().ClusterConfig
-	}
-}
-
-func onAddOrUpdateProjects(obj interface{}, projectMap map[string]*config.Project) map[string]*config.Project {
-	configMap := obj.(*v1.ConfigMap)
-	projectJSONString, ok := configMap.Data["project"]
+	dataJSONString, ok := configMap.Data["data"]
 	if !ok {
-		_ = helpers.Logger.LogError(helpers.GetRequestID(context.TODO()), "Unable to watch kube store project as field (cluster) not found in config map", nil, nil)
-		return nil
+		_ = helpers.Logger.LogError(helpers.GetRequestID(context.TODO()), fmt.Sprintf("%s event occured on resource config map, but (resource) field was not found in config map data", eventType), nil, nil)
+		return "", "", nil
 	}
 
-	v := new(config.Project)
-	if err := json.Unmarshal([]byte(projectJSONString), v); err != nil {
-		_ = helpers.Logger.LogError(helpers.GetRequestID(context.TODO()), "Unable to unmarshal config map data while watching kube store project", nil, nil)
-		return nil
+	var v map[string]interface{}
+	if err := json.Unmarshal([]byte(dataJSONString), &v); err != nil {
+		_ = helpers.Logger.LogError(helpers.GetRequestID(context.TODO()), "Unable to unmarshal resource config map data while watching kube store project", nil, map[string]interface{}{"resourceId": resourceID, "eventType": eventType})
+		return "", "", nil
 	}
-	projectMap[v.ID] = v
-	return projectMap
+	return eventType, resourceID, v
 }
 
-// WatchAdminConfig maintains consistency over all projects
-func (s *KubeStore) WatchAdminConfig(cb func(clusters []*config.Admin)) error {
+// WatchResources maintains consistency over all projects
+func (s *KubeStore) WatchResources(cb func(eventType, resourceID string, resource interface{})) error {
```

Comment:
I suggest you take the resource type from the label and pass that too in the callback. Lets not rely too heavily on the resourceID string to retrieve the resource type